src/pint_pal/noise_utils.py

```diff
@@ -43,214 +130,477 @@ def analyze_noise(chaindir = './noise_run_chains/', burn_frac = 0.25, save_corne
+                    histtype="step",
+                    color="orange",
+                    label="Past",
+                )
+                ax.axvline(
+                    chain_compare[:, idx][mp_compare_idx], ls="--", color="orange"
+                )
+            if "_wb" in chaindir:
+                ax.set_xlabel(par, fontsize=8)
+            else:
+                ax.set_xlabel(par, fontsize=10)
             ax.set_yticks([])
             ax.set_yticklabels([])
 
-            if j == (nrows*ncols)-1 or idx == len(pars_short)-1:
+            if j == (nrows * ncols) - 1 or idx == len(pars_short) - 1:
                 pl.tight_layout()
                 pl.savefig(f"{figbase}_{pp}.pdf")
 
         # Wasn't working before, but how do I implement a legend?
-        #ax[nr][nc].legend(loc = 'best')
+        # ax[nr][nc].legend(loc = 'best')
         pl.show()
     
-    ml_idx = np.argmax(chain[burn:, -4])
-
-    wn_vals = chain[burn:, :-4][ml_idx]
-
-    wn_dict = dict(zip(pars, wn_vals))
-
-    #Print bayes factor for red noise in pulsar
-    rn_bf = model_utils.bayes_fac(chain[burn:, -5], ntol=1, logAmax=-11, logAmin=-20)[0]
-
-    return wn_dict, rn_bf
-
-def model_noise(mo, to, vary_red_noise = True, n_iter = int(1e5), using_wideband = False, resume = False, run_noise_analysis = True, wb_efac_sigma = 0.25, base_op_dir = "./"):
+    noise_dict = noise_core.get_map_dict()
+
+    # Print bayes factor for red noise in pulsar
+    rn_amp_nm = psr_name+"_red_noise_log10_A"
+    rn_bf = model_utils.bayes_fac(noise_core(rn_amp_nm), ntol=1, logAmax=-11, logAmin=-20)[0]
+
+    return noise_core, noise_dict, rn_bf
+
+
+def model_noise(
+    mo,
+    to,
+    using_wideband=False,
+    resume=False,
+    run_noise_analysis=True,
+    wb_efac_sigma=0.25,
+    base_op_dir="./",
+    model_kwargs={},
+    sampler_kwargs={},
+    return_sampler=False,
+):
     """
-    Setup enterprise PTA and perform MCMC noise analysis
+    Setup enterprise or discovery likelihood and perform Bayesian inference on noise model
 
     Parameters
     ==========
     mo: PINT (or tempo2) timing model
     to: PINT (or tempo2) TOAs
-    red_noise: include red noise in the model
-    n_iter: number of MCMC iterations; Default: 1e5; Recommended > 5e4
     using_wideband: Flag to toggle between narrowband and wideband datasets; Default: False
+    resume: Flag to resume overwrite previous run or not.
     run_noise_analysis: Flag to toggle execution of noise modeling; Default: True
+    noise_kwargs: dictionary of noise model parameters; Default: {}
+    sampler_kwargs: dictionary of sampler parameters; Default: {}
+    return_sampler: Flag to return the sampler object; Default: False
+    
+    Recommended to pass model_kwargs and sampler_kwargs from the config file.
+    Default kwargs given by function `get_model_and_sampler_default_settings`.
+    Import configuration parameters:
+        likelihood: choose from ['Enterprise', 'discovery']
```

Comment:
Something to note is to check capitalization for `"Enterprise"`

~ Joe G.

Reply:
thanks for pointing this out!
every instance of `enterprise` is now lower case.